### PR TITLE
Propagate editor UI state transitions to add-ons

### PR DESCRIPTION
**Pull request by glutanimate**
Signal changes to the editor interface state via a Python hook, allowing add-on subscribers to adjust their UI to state transitions.

### CODE REVIEW

File: ts/editor/NoteEditor.svelte [hunk 1]
```diff
@@ -526,6 +565,13 @@ License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
             ...oldEditorAdapter,
         });
 
+        editorState = getEditorState(
```
Comment (glutanimate) [inline ts/editor/NoteEditor.svelte:568]: Not sure if the repetition here can be avoided somehow?

File: ts/editor/NoteEditor.svelte [hunk 2]
```diff
@@ -485,6 +486,44 @@ License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
         return false;
     }
 
+    // Signal editor UI state changes to add-ons
+
+    let editorState: EditorState = EditorState.Initial;
+    let lastEditorState: EditorState = editorState;
+
+    function getEditorState(
+        ioMaskEditorVisible: boolean,
```
Comment (glutanimate) [inline ts/editor/NoteEditor.svelte:495]: Essentially following the current render logic here. We'll probably be able to simplify this in the future (#2624)

File: ts/editor/NoteEditor.svelte [hunk 3]
```diff
@@ -485,6 +486,44 @@ License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
...
+        } else if (!ioMaskEditorVisible && isImageOcclusion) {
+            return EditorState.ImageOcclusionFields;
+        }
+        return EditorState.Fields;
+    }
+
+    function signalEditorState(newState: EditorState) {
+        tick().then(() => {
+            globalThis.editorState = newState;
+            bridgeCommand(`editorState:${newState}:${lastEditorState}`);
```
Comment (glutanimate) [inline ts/editor/NoteEditor.svelte:513]: In the future we might want to add a proper API/hook for JS subscribers here.  Did not pursue this further due to lack of time and as this solution works well for IOE for now.

File: qt/aqt/editor.py [hunk 4]
```diff
@@ -124,6 +136,7 @@ def __init__(
         self.last_field_index: int | None = None
         # current card, for card layout
         self.card: Card | None = None
+        self.state: EditorState | None = None
```
Comment (dae) [inline qt/aqt/editor.py:139]: Do we need both None and INITIAL states here?

**Review by dae — COMMENTED** [review]
Looks fine, just a quick question.

File: qt/aqt/editor.py (hunk 4 above)
Comment (glutanimate) [reply qt/aqt/editor.py]: No, you're right, don't see much use in starting with `None`. Changed!

**Pull request merged**